### ISSUE TITLE
Cannot use variable in filter's rule

### ISSUE THREAD

**Issue by it-management** [roundcube/roundcubemail#5778]
If I try to use variable in filter's rule this way:
Actual date - date (yyyy-mm-dd) - value is greater or equal - ${date_variable}
it will not be validated, because a date or a part of date is expected.

If I directly edit .SIEVE file, everything works and I can see it in rule definition, but it's impossible to save it still.

**Comment by alecpl** [comment]
Yeah, we should probably skip format checking if the field contains `$` character. Internally it might be more complicated.

**Comment by leofeyer** [comment]
We are having the same problem:

<img width="867" src="https://user-images.githubusercontent.com/1192057/57508659-61d52c00-7302-11e9-9af9-b02a4885b0c2.png">
